langgraph & gemini pooling add

choijhyeok opened the pull request:
- README.md 에 mermaid flow 추가
- gemini_pooling 기법 기존 석현님 Langgraph에 적용
- repo 구조 변경
   - 개인 test폴더는 test/ 하위에 넣는 방식으로 적용

choijhyeok committed:
langgraph & gemini pooling add

choijhyeok committed:
UI and pipeline 개발

maylilyo reviewed (approved):
- 코드 병합 내용, Environment Setting 및 README File 내 변경사항을 확인하였습니다.
- pipeline_ui의 코드들에 대해 모임 내 간략한 리뷰가 필요합니다.
- readme의 240 line부터 명시적으로 이름이 들어가는데, 상대경로로 바꾸지 않으셔도 괜찮으실까요?

maylilyo commented:
석현님 코드 진척이 많이 되어있는 상황이셔서, 코드 머징이 가능한 상황인지 확인 부탁드립니다. @srrk-GreenMan

choijhyeok committed:
feat: Add pipeline diagram and node state management

- Introduced a new PipelineDiagram component to visualize the processing stages of job items.
- Enhanced JobItemCard to display node states and results dynamically.
- Updated process_single_item to track and report node states and results during processing.
- Added new types for NodeState, NodeStates, and NodeResults to manage node-specific data.
- Implemented animations for data flow and node interactions in the UI.
- Created a new JSON checkpoint file to store processing results and QA data.
- Added synthetic table and QA results to the output JSON structure for better data representation.

srrk-GreenMan commented:
이제 퇴근해서 막 pull request 날리려고 했는데 이미 많이 작업해주셨군요.. 

https://github.com/Pseudo-Lab/TableMagnifier/tree/feature/agenticmoduletestcase-issue

수정된 내용을 보았을 땐 제가 작업한 내용과 유사한 것 같습니다. 재혁님이 수정한 방향으로 가도 괜찮을 것 같아요

다만 pipeline ui는 약간 고민이긴 합니다. 제가 처음에 작업하려고 했던 프론트엔드는 pipeline은 QA 등이 마음에 들지 않을 때 데이터 검수를 위한 것이었는데 지금은 데이터 생성을 위한 ui인 것 같아서 그냥 배치 단위로 실행시켜서 돌리고 검수를 프론트로 하는 게 좋지 않을까 싶습니다...!


cc: @choijhyeok @maylilyo

choijhyeok commented:
@maylilyo 

240 line 사실 그냥 제가쓰던 거 붙여넣다보니 제 경로가 들어갔네여 이건 수정하겠습니다. 
(그냥 backend , frontend 들가서 npm install , npm dev하면 된다는걸 표현한거였습니다.)

pipeline_ui 같은경우는 기존 석현님이 작업하신 방향성을 살려서 batch 처리, 진행상황 (langgraph node 단계를 보여주는 UI) 정도만 추가한 형태입니다. 검수를 위한 UI 로 바꾸는 형상이면 조금 차이가 있을거 같습니다.

+ 추가로 merge는 사실 누가 먼저 부어도 상관이 없습니다.  (코드 merge 상황 이야기가 나와서 첨언드리면)
단 PR을 통해 review가 끝났을때 merge를 선행한 사람이 있다고 알려주면 기존 PR 날린사람은 git pull 을 받아서 (dev -> branch ) 를 통한 최신화를 수행하고 변경사항을 다시 올리면 됩니다. (git에 익숙한 사람일 경우 본인의 branch를 rebase 쳐서 맨위로 올린다음 git pull 받아도됨)

++ 사실 이 부분때문에 무료 slack을 쓰면 webhook으로 서로 누가 git merge 했는지를 알수있어서 일반적으로 code merge 방지를 위해서 web hook을 많이 사용하는데 현 상황에서는 merge한 사람이 카톡에 말하지 않으면 꼬여버릴거 같긴 합니다..

choijhyeok committed:
readme 절대경로 삭제

choijhyeok committed:
gitignore 추가

choijhyeok committed:
chore: remove tracked files now in .gitignore

choijhyeok committed:
chore: remove dataset folder now in .gitignore

choijhyeok committed:
Merge branch 'dev' of https://github.com/Pseudo-Lab/TableMagnifier into feature/repo-정리-및-기능-병합-issue

choijhyeok committed:
dev all merge
